### DISCUSSION TITLE
Deploying multiple production environments from one  branch / one repo

### DISCUSSION THREAD

**Discussion by janus-reith** [vercel/vercel#4879] [Help] [answered]
Currently, I have three production environments, which all are build from the same master branch but have their own environment variables both during build and runtime.

I would like to avoid creating three separate repositories.

How could I deploy these on vercel properly? The Github Integration restricts the repository to be linked to one vercel project.
I would be ok with splitting up the 3 deploymens into 3 separate production branches if that is necessary, but I'm wondering how that works out with the 3 environments production/staging/development which vercel offers per project.
As there also is the option to include a `vercel.json` file in the branch, could I use .`env `+ `build.env` in there and ignore the project-wide configuration, or would they overwrite each other?

I appreciate suggestions, maybe there are other options to consider aswell.

**Comment by janus-reith** [comment]
According to https://github.com/vercel/vercel/discussions/4719#discussioncomment-35580 there currently is no way for this while keeping the git integration intact, but hopefully there is soon.

**Comment by janus-reith** [comment, marked as answer]
This feature was added in the meantime.

**Reply by arnaudriegert** [reply]
Hi @janus-reith, which feature are you referring to?

The issue you referenced above mentions that we can deploy different apps from a same repo, but from what I understand, you're looking for a way to deploy the same app in multiple production environments (with different environment variables).

If there's a way to do that, I have the same need :) 

Thank you!

**Reply by janus-reith** [reply]
Hey @arnaudpeich back when I opened that ticket I could only link my github repository with one vercel project.

That was changed, so now I can have multiple projects on vercel all referencing the same github repo.
(Also meaning, a push to the main branch of that repo would rebuild all vercel projects referring it).

As I can define my environment variables separately per vercel project, that works fine.

**Reply by arnaudriegert** [reply]
Ok thanks a lot! The issue I had with this method was that it would deploy one preview app _per project_ for each commit on non-master branches, but in the meantime I found that there were ways to prevent that, so it should do the trick :)

**Reply by baezanat** [reply]
@arnaudpeich How were you able to stop Vercel from deploying one preview app per project for each commit on non-master branches? I'm facing the same issue.  :)

**Reply by avarun42** [reply]
@baezanat for any Vercel project you don't want previews to be deployed for, go into Settings -> Git -> Ignored Build Step and set the value as `[ "$VERCEL_ENV" != production ]`.

**Reply by ManuLpz4** [reply]
Sorry but I want to deploy a preview env and a dev env on every develop branch push. How can I achieve it with one Vercel project?

**Reply by ackvf** [reply]
You can have multiple domains in one vercel project, each assigned to a different branch, simulating multiple "production" apps, but internally, Vercel will still keep the other non-production deployments as "Preview". That includes the Vercel Toolbar, which can be also disabled for whole project or by using [headers](https://vercel.com/docs/workflow-collaboration/vercel-toolbar#disable-toolbar-for-session).

**Reply by shubasishdas** [reply]
How can I deploy multiple branch as production rather preview?

**Reply by cobra0411** [reply]
is there any way to deploy different branches of a project on different domains. production only not in preview mode?

**Reply by 0x33dm** [reply]
> is there any way to deploy different branches of a project on different domains. production only not in preview mode?

also looking for that without having to dupiclate my project on vercel.

**Reply by manokaran68** [reply]
Same requirement here. Currently I have v1.mydomain.com as prod on branch main. I want to deploy v2.mydomain.com also as prod but on branch beta so that when everything works as expected, I can remove current prod and rename v2 to v1. Is that possible in Vercel?

**Reply by 0x33dm** [reply]
> Same requirement here. Currently I have v1.mydomain.com as prod on branch main. I want to deploy v2.mydomain.com also as prod but on branch beta so that when everything works as expected, I can remove current prod and rename v2 to v1. Is that possible in Vercel?

that's a very good example. apparently one of the suggestions by support is to disable authentication on the preview branches, but that sounds like a bit of an overkill
